### DISCUSSION TITLE
Frame 47:6 - Unable to run the extended sum

### DISCUSSION THREAD

**Discussion by sreeramcoder** [themetaschemer/malt#36] [Q&A] [answered]
Hello,

I am unable to get below line (present in frame 47:6) working in my machine.

`
(+ [2] [7])
`
I see that '+' behaves like a normal operator and does not descend into the tensor, as explained in frame 48:7.

I tried different approaches but no luck yet. here is the complete source code with the errors I got in comments.

```
`#lang racket
(require malt)

; Input #1

(display (+ [2] [3]))

;application: not a procedure;
; expected a procedure that can be applied to arguments
;  given: 2
;  context...:
;   /home/sreeram/prog/ll/test.rkt:4:0
;   body of "/home/sreeram/prog/ll/test.rkt"


; Input #2

(display (+ (tensor '(2)) (tensor '(3))))

;vector-length: contract violation
;  expected: vector?
;  given: '(2)
;  context...:
;   /home/sreeram/.racket/8.3/pkgs/malt/learner/tensors/D-extend.rkt:30:2: desc
;   /usr/share/racket/collects/racket/vector.rkt:62:0: vector-map/update
;   /usr/share/racket/collects/racket/vector.rkt:106:0: vector-map
;   /home/sreeram/prog/ll/test.rkt:6:0
;   body of "/home/sreeram/prog/ll/test.rkt"
`
```

My racket version is v8.3.

**Comment by sreeramcoder** [comment]
Please note that I am new to racket (have some idea about scheme after stumbling half way thru SICP). So please bear with me, if I have overlooked any obvious step & guide me towards the light :).

**Comment by themetaschemer** [comment, marked as answer]
Hi Sreeram, 

Please refer to the section on Transcribing to Scheme on page xxiii to convert our notations into Scheme. Unfortunately, we had to abbreviate some of our notations so that the programs fit properly within our little boxes.

In your example, you'll have to say: `(+ (tensor 2) (tensor 3))` ... for bigger tensors `(+ (tensor 1 2 3 4) (tensor 5 6 7 8))`.

**Reply by sreeramcoder** [reply]
Thank you so much, it helped !

**Comment by alexgian** [comment]
```
> (display (+ #[1] #[2]))
(tensor 3)
```
works as a q&d shorthand, too.  I don't know if it's naughty, but I find I use it all the time when messing around.
